napari/layers/base/base.py

```diff
@@ -1659,14 +1659,14 @@ def _world_to_layer_dims_impl(
         """
         Static for ease of testing
         """
-        world_dims = np.asarray(world_dims)
-        assert world_dims.min() == 0
-        assert world_dims.max() == len(world_dims) - 1
-        assert world_dims.ndim == 1
         offset = ndim_world - ndim
-        order = world_dims - offset
-        order = order[order >= 0]
-        return order - order.min()
+        order = np.array(world_dims)
+        if offset == 0:
+            return order
+        if offset < 0:
+            return np.concatenate((np.arange(-offset), order - offset))
```

Comment:
A comment here would be great, explaining what offset is, I find this  abit hard to follow :P

Reply:
Thanks for giving this a check! While working on the comment and rechecking things locally I noticed that this fails when trying to do the point selection after changing the viewer to 2D :( Will mark this as a draft for the moment and keep checking